src/quantuminspire/qiskit/backend_qx.py

```diff
@@ -255,18 +261,35 @@ def _generate_cqasm(self, experiment: QasmQobjExperiment, measurements: Measurem
         with io.StringIO() as stream:
             stream.write('version 1.0\n')
             stream.write('# cQASM generated by QI backend for Qiskit\n')
-            stream.write('qubits %d\n' % number_of_qubits)
+            stream.write(f'qubits {number_of_qubits}\n')
             for instruction in instructions:
                 parser.parse(stream, instruction)
             return stream.getvalue()
 
+    def generate_user_data(self, experiment: QasmQobjExperiment, measurements: Measurements) -> Dict[str, Any]:
+        """
+        Generates the user_data for this experiment. The user_data is saved together with the job and consists of
+        data that is necessary to process the result of the experiment correctly.
+
+        :param experiment: The experiment that contains the header information to save in the user data.
+        :param measurements: The measurement instance containing measurement information and measurement functionality.
+
+        :return:
+            A structure with user data that is needed to process the result of the experiment.
+        """
+        user_data = {'name': experiment.header.name, 'metadata': serializer.serialize(experiment.header.metadata),
+                     'qubit_labels': experiment.header.qubit_labels, 'qreg_sizes': experiment.header.qreg_sizes,
+                     'clbit_labels': experiment.header.clbit_labels, 'creg_sizes': experiment.header.creg_sizes,
+                     'global_phase': experiment.header.global_phase, 'memory_slots': experiment.header.memory_slots,
+                     'measurements': measurements.to_dict()}
+        return user_data
```

Comment:
nit: what is the reason we don't just inline the return? The assignment of the local variable is never used besides returning it immediately.

Reply:
None actually. I will change it